front/types/assistant/actions/retrieval.ts

```diff
@@ -3,10 +3,20 @@
  */
 
 import { ModelId } from "@app/lib/databases";
+import { ioTsEnum } from "@app/lib/utils";
 import { AgentActionConfigurationType } from "@app/types/assistant/agent";
 import { AgentActionType } from "@app/types/assistant/conversation";
 
-export type TimeframeUnit = "hour" | "day" | "week" | "month" | "year";
+export const TIME_FRAME_UNITS = [
+  "hour",
+  "day",
+  "week",
+  "month",
+  "year",
+] as const;
+export type TimeframeUnit = (typeof TIME_FRAME_UNITS)[number];
```

Comment:
nit: uniformization => if it's TimeFrame then TimeFrameUnit (or change TimeFrame to Timeframe)

Reply:
Good point thanks! Willfix